libmachine/mcndockerclient/docker_client.go

```diff
@@ -1,46 +1,61 @@
 package mcndockerclient
 
 import (
+	"context"
 	"fmt"
+	"net/http"
 
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/client"
 	"github.com/rancher/machine/libmachine/cert"
-	"github.com/samalba/dockerclient"
 )
 
 // DockerClient creates a docker client for a given host.
-func DockerClient(dockerHost DockerHost) (*dockerclient.DockerClient, error) {
+func DockerClient(dockerHost DockerHost) (*client.Client, error) {
 	url, err := dockerHost.URL()
 	if err != nil {
 		return nil, err
 	}
 
 	tlsConfig, err := cert.ReadTLSConfig(url, dockerHost.AuthOptions())
 	if err != nil {
-		return nil, fmt.Errorf("Unable to read TLS config: %s", err)
+		return nil, fmt.Errorf("unable to read TLS config: %s", err)
 	}
 
-	return dockerclient.NewDockerClient(url, tlsConfig)
+	httpClient := &http.Client{
```

Comment:
I think we need to set timeout or is set by default? The earlier behavior set it to 30s.